src/types.md

```diff
@@ -506,10 +510,10 @@ Trait objects implement the base trait, its auto traits, and any super traits
 of the base trait.
 
 Trait objects are written as the path to the base trait followed by the list
-of auto traits followed optionally by a lifetime bound all separated by `+`. For
-example, given a trait `Trait`, the following are all trait objects: `Trait`,
-`Trait + Send`, `Trait + Send + Sync`, `Trait + 'static`,
-`Trait + Send + 'static`.
+of auto traits followed optionally by a lifetime bound all separated by and
+optionally terminated by  `+`. For example, given a trait `Trait`, the following
+are all trait objects: `Trait`, `Trait + Send`, `Trait + Send + Sync`,
+`Trait + 'static`, `Trait + Send + 'static`, `Trait +`.
```

Comment:
`'static + Send + Trait` also works.
In general, trait objects has exactly the same syntax as bounds after `T` in `fn f<T: BOUNDS>() { ... }`, it would be great to unify them in the reference as well.
Trait objects just have some additional semantic restrictions (n(lifetimes) <=1, n(non-auto-traits) <=1).

Reply:
`'static + Send + Trait` explicitly doesn't work giving me `error[E0225]: only auto traits can be used as additional traits in a trait object`. `'static + Trait` does work.

Generic bounds don't seem to be in the reference at all right now. 🙁

Reply:
I'm working on it. :shades:

Reply:
I'm working on adding some documentation for them soon.